akka-http-circe/src/main/scala/de/heikoseeberger/akkahttpcirce/CirceSupport.scala

```diff
@@ -67,9 +72,7 @@ trait CirceSupport {
     */
   implicit def circeToEntityMarshaller[A](
       implicit encoder: Encoder[A],
-      printer: Json => String = Printer.noSpaces.pretty
+      printer: Json ⇒ String = Printer.noSpaces.pretty
   ): ToEntityMarshaller[A] =
-    Marshaller.StringMarshaller
-      .wrap(`application/json`)(printer)
-      .compose(encoder.apply)
+    jsonStringMarshaller.compose(printer).compose(encoder.apply)
```

Comment:
Thanks for the explanation. I already did get your point about possibly improving performance via factoring out the marshallers. What I'd like to understand is the difference btw. `Marshaller.stringMarshaller(`application/json`)` and `Marshaller.StringMarshaller.wrap(`application/json`)`.